src/pages/portfolio/ui/assets-table.tsx

```diff
@@ -0,0 +1,305 @@
+                    <div className='flex items-center gap-2'>
+                      <div className='w-8 h-8'>
+                        <Skeleton />
+                      </div>
+                      <div className='w-20 h-4'>
+                        <Skeleton />
+                      </div>
+                    </div>
+                  </Table.Td>
+                  <Table.Td>
+                    <div className='w-24 h-4'>
+                      <Skeleton />
+                    </div>
+                  </Table.Td>
+                  <Table.Td>
+                    <div className='w-24 h-4'>
+                      <Skeleton />
+                    </div>
+                  </Table.Td>
+                  <Table.Td>
+                    <div className='w-24 h-4'>
+                      <Skeleton />
+                    </div>
+                  </Table.Td>
+                  <Table.Td>
+                    <div className='flex gap-2'>
+                      <div className='w-5 h-5'>
+                        <Skeleton />
+                      </div>
+                      <div className='w-5 h-5'>
+                        <Skeleton />
+                      </div>
+                    </div>
+                  </Table.Td>
+                </Table.Tr>
+              ))}
+            </Table.Tbody>
+          </Table>
+        </Density>
+      </Card>
+    </div>
+  );
+};
+
+const NotConnectedNotice = () => {
+  return (
+    <div className='p-6'>
+      <Card>
+        <div className='flex flex-col items-center justify-center h-[400px] gap-4'>
+          <Text color='text.secondary' small>
+            Connect wallet to see your assets
+          </Text>
+        </div>
+      </Card>
+    </div>
+  );
+};
+
+const calculateAssetDistribution = (balances: BalancesResponse[]) => {
+  const validBalances = balances.filter(balance => {
+    const valueView = getBalanceView.optional(balance);
+    const metadata = getMetadataFromBalancesResponse.optional(balance);
+    return valueView && metadata;
+  });
+
+  // Calculate values first
+  const valuesWithMetadata = validBalances.map(balance => {
+    const valueView = getBalanceView.optional(balance);
+    const metadata = getMetadataFromBalancesResponse.optional(balance);
+    const amount = valueView?.valueView.value?.amount;
+    if (!amount || !metadata) {
+      return { value: 0, balance };
+    }
+    const formattedAmount = Number(
+      formatAmount({
+        amount,
+        exponent: getDisplayDenomExponent(metadata),
+      }),
+    );
+
+    return {
+      value: formattedAmount,
+      balance,
+    };
+  });
+
+  const totalValue = valuesWithMetadata.reduce((acc, { value }) => acc + value, 0);
+
+  if (totalValue === 0) {
+    return { distribution: [], sortedBalances: [] };
+  }
+
+  const distributionWithMetadata = valuesWithMetadata.map(({ value, balance }) => ({
+    percentage: (value / totalValue) * 100,
+    color: `hsl(${Math.random() * 360}, 70%, 50%)`,
```

Comment:
Can we make the palette selection deterministic based on the asset id? i think we might not need any other tweaks to it because it's uniformly distributed, otherwise we will adjust